contrib/babelfishpg_tsql/src/pl_exec-2.c

```diff
@@ -4182,17 +4184,38 @@ exec_stmt_partition_function(PLtsql_execstate *estate, PLtsql_stmt_partition_fun
 				errmsg("There is already an object named '%s' in the database.", partition_function_name)));
 	}
 
-	/* get the tsql typename from the input type */
+	/*
+	 * Try to find the TSQL type name for the input type and if it fails
+	 * and input type is DOMAIN type created in sys schema then
+	 * find the TSQL type name using the base type of DOMAIN.
+	 */
 	InitFunctionCallInfoData(*fcinfo, NULL, 0, InvalidOid, NULL, NULL);
 	fcinfo->args[0].value = ObjectIdGetDatum(typ->typoid);
 	fcinfo->args[0].isnull = false;
 	tsql_type_datum = (*common_utility_plugin_ptr->translate_pg_type_to_tsql) (fcinfo);
-
 	if (tsql_type_datum)
 	{
 		tsql_typename = text_to_cstring(DatumGetTextPP(tsql_type_datum));
 	}
-
+	else
+	{
+		tuple = SearchSysCache1(TYPEOID, ObjectIdGetDatum(typ->typoid));
```

Comment:
Should add check for tuple : 
```
		if (!HeapTupleIsValid(tup))
			??
```

Reply:
typ->typoid will always be valid OID and will always have entry in pg_type. Please check the definition of parse_datatype for more details.

Reply:
What will happen if customer delete that from pg_type manually.

Reply:
If type is deleted, parse_datatype API in ANTLR itself will raise error.